torch/csrc/StorageMethods.cpp

```diff
@@ -173,18 +173,16 @@ static PyObject* THPStorage_fromBuffer(
   PyObject* dtype_obj = nullptr;
   c10::ScalarType scalar_type = at::kByte;
   Py_buffer buffer = {};
-  // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays,clang-diagnostic-writable-strings)
-  static char* kwlist[] = {
+  // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays)
+  constexpr char* kwlist[] = {
       "buffer", "byte_order", "count", "offset", "dtype", nullptr};
```

Comment:
Just curious, why not use `std::array`, while you are at it
```suggestion
  constexpr std::array<char*> kwlist = {
      "buffer", "byte_order", "count", "offset", "dtype", nullptr};
```

Reply:
std::array can't deduce size automatically and we have to specify it. But the main reason is to avoid change more parts of the old code.

Reply:
And I think it is not bad to use C types for C API